astroquery/simbad/core.py

```diff
@@ -392,7 +392,47 @@ def query_objects(self, object_names, wildcard=False, verbose=False):
         table : `~astropy.table.Table`
             Query results table
         """
-        return self.query_object('\n'.join(object_names), wildcard)
+        result = self.query_object('\n'.join(object_names), wildcard)
+
+        if keep_input:
+            return self.add_input_column_to_simbad_result(object_names, verbose=verbose)
+        else:
+            return result
+
+
+    def add_input_column_to_simbad_result(self, input_list, verbose=False):
+        """
+        Adds 'INPUT' column to the result of a Simbad query
+
+        Parameters
+        ----------
+        object_names : sequence of strs
+                names of objects input to most recent query
+        verbose : boolean, optional
+            When `True`, verbose output is printed
+
+        Returns
+        -------
+        table : `~astropy.table.Table`
+            Query results table
+        """
+        error_string = self.last_parsed_result.error_raw
+        fails = []
+
+        for error in error_string.split("\n"):
+            start_loc = error.rfind(":") + 2
+            fail = error[start_loc:]
+            fails.append(fail)
+
+        successes = [s for s in input_list if s not in fails]
+        if verbose:
+            say = "There were {} successful Simbad matches and {} failures."
+            print say.format(len(successes), len(fails))
```

Comment:
`print` statements are not permitted in astroquery code; instead use `log.info` here.